src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,58 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
+import { Method } from '../../types/Method';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  setFilterMethod: (method: string) => void,
+  query: string,
+  setQuery: (quary :string) => void,
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<Props> = ({
+  setFilterMethod,
+  query,
+  setQuery,
+}) => {
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            onChange={event => setFilterMethod(event.target.value)}
+          >
+            <option value={Method.all}>All</option>
+            <option value={Method.active}>Active</option>
+            <option value={Method.completed}>Completed</option>
```

Comment:
will be better use `.map()` for iterate

Reply:
I can do it, but... I mean, is it really necessary? I will need to create a new array for it, and copy all the methods in it.